### PR TITLE
Allow token creation from an existing payload

### PR DESCRIPTION
**Pull request by RicoLannez**
Add virtual method in JWTTokenManagerInterface to allow token creation from an existing payload

### CODE REVIEW

**Comment by RicoLannez** [comment]
Hi, as discussed in this [PR](https://github.com/lexik/LexikJWTAuthenticationBundle/pull/609) it would be great to be able to create an AccessToken from an existing payload.

File: Services/JWTTokenManagerInterface.php [hunk 1]
```diff
@@ -10,6 +10,8 @@
  * JWT tokens.
  *
  * @author Robin Chalas <robin.chalas@gmail.com>
```
Comment (chalasr) [inline Services/JWTTokenManagerInterface.php:12]: Please add yourself to authors, you extended it :)

File: Services/JWTManager.php [hunk 2]
```diff
@@ -88,6 +88,39 @@ public function create(UserInterface $user)
         return $jwtString;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function createFromPayload(UserInterface $user, array $payload)
+    {
+        $payload = array_merge(['roles' => $user->getRoles()], $payload);
+        $this->addUserIdentityToPayload($user, $payload);
```
Comment (chalasr) [inline Services/JWTManager.php:97]: Can you reuse the duplicated logic below by moving it to a private method called from both `create()` and `createFromPayload()`?

**Review by chalasr — COMMENTED** [review]
2 minor comments, looks good to me otherwise. Thanks for the PR @RicoLannez

**Review by chalasr — APPROVED** [review]
I will merge/release tonight.

**Comment by chalasr** [comment]
Thank you @RicoLannez.

**Pull request closed without merging**